pcap-openvizsla.c

```diff
@@ -0,0 +1,199 @@
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "pcap-int.h"
+#include "pcap-openvizsla.h"
+
+#include <ov.h>
+
+#define OPENVIZSLA_IFACE "openvizsla"
+
+struct pcap_openvizsla {
+	enum ov_usb_speed usb_speed;
+	struct ov_device* ov;
+};
+
+struct pcap_openvizsla_read {
+	pcap_handler callback;
+	u_char *user;
+};
+
+static void openvizsla_read_cb(struct ov_packet* packet, void* data) {
+	struct pcap_openvizsla_read* r = (struct pcap_openvizsla_read*)data;
+
+	struct pcap_pkthdr pkth;
+	pkth.caplen = ov_to_host_16(packet->size) + sizeof(struct ov_packet);
+	pkth.len = pkth.caplen;
+	pkth.ts.tv_sec = 0;
+	pkth.ts.tv_usec = 0;
+
+	r->callback(r->user, &pkth, (const u_char*)packet);
+}
+
+static int openvizsla_read(pcap_t *handle, int max_packets, pcap_handler callback, u_char *user)
+{
+	struct pcap_openvizsla* handlep = handle->priv;
+	struct pcap_openvizsla_read r;
+
+	r.callback = callback;
+	r.user = user;
+
+	int ret = 0;
+
+	if ((ret = ov_capture_start(handlep->ov, handle->buffer, handle->bufsize, openvizsla_read_cb, &r)) < 0) {
+		snprintf(handle->errbuf, PCAP_ERRBUF_SIZE,
+			"Can't start capture: %s", ov_get_error_string(handlep->ov));
+		return PCAP_ERROR;
+	}
+
+	if ((ret = ov_capture_dispatch(handlep->ov, max_packets)) < 0) {
```

Comment:
Is there a way in which the callback routine can break out of the loop?  If, for example, a callback routine calls `pcap_breakloop()`, that should cause `pcap_loop()` or `pcap_breakloop()` not to process any more packets.

Reply:
`breakloop_op` implementation has been added.